### PR TITLE
Add Iterators and the for..of loop

### PR DESCRIPTION
**Pull request by Elchi3**
(For Iterator methods, see https://github.com/web-platform-dx/web-features/pull/1656)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
https://chromestatus.com/feature/4696563918045184

Using `compute_from` here, as I think the `for_of` key is the most relevant data point for this one.

There are also iterator symbols on many JS builtins and these could belong to this feature (e.g. `javascript.builtins.String.@@iterator `and many more). `Array` has its own feature: https://github.com/web-platform-dx/web-features/blob/main/features/array-iterators.yml. Not sure if this should rather all be here or if we want this to be separate for each builtin.

### CODE REVIEW

File: features/iterators.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Iterators and the for..of loop
```
Comment (autonome) [inline features/iterators.yml:1]: s/../.../

File: features/iterators.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+description: The `for...of` loop operates on a sequence of values sourced from an iterable object (arrays, array-like objects, DOM collections, iterators, generators, and user-defined iterables).
```
Comment (autonome) [inline features/iterators.yml:2]: non-blocking: I prefer the "such as..." vs a very long parenthetical

**Review by autonome — DISMISSED** [review]

File: features/iterators.yml (hunk 2 above)
Comment (autonome) [reply features/iterators.yml:2]: As the feature covers both, should mention `Iterator` explicitly here.

**Comment by autonome** [comment]
> There are also iterator symbols on many JS builtins and these could belong to this feature (e.g. `javascript.builtins.String.@@iterator `and many more). `Array` has its own feature: https://github.com/web-platform-dx/web-features/blob/main/features/array-iterators.yml. Not sure if this should rather all be here or if we want this to be separate for each builtin.

I think is good how you've done this as the generic + usage (`for...of`), and leaving instance-specific iterators w/ their instances.

Spec authors and implementers may think of something like iterableness as group that crosses instances of it, but developers think on the instance/usage level. Eg, imagine you're looking up if `something` is iterable - developer would go to the `something` docs for that.

File: features/iterators.yml (hunk 2 above)
Comment (Elchi3) [inline features/iterators.yml:2]:
```suggestion
description: The `for...of` loop operates on a sequence of values sourced from an iterable object, such as arrays, array-like objects, DOM collections, iterators, generators, and user-defined iterables. All built-in iterators inherit from the `Iterator` class. 
```

**Review by ddbeck — APPROVED** [review]

**Pull request merged**